File: integration-test/src/main/java/com/sequenceiq/it/spark/docker/model/InspectContainerResponse.java
package com.sequenceiq.it.spark.docker.model;

import com.google.gson.annotations.SerializedName;

import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;

public class InspectContainerResponse {

    @SerializedName("Id")
    private String id;

    @SerializedName("State")
    private ContainerState state;

    public InspectContainerResponse() {
    }

    @SuppressFBWarnings("URF_UNREAD_FIELD")
    public InspectContainerResponse(String id) {
        this.id = id;
        state = new ContainerState();
        state.setRunning(true);
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setState(ContainerState state) {
        this.state = state;
    }

    public static class ContainerState {

        @SerializedName("Running")
        private boolean running;

        public boolean isRunning() {
            return running;
        }

        public void setRunning(boolean running) {
            this.running = running;
        }
    }

}

File: integration-test/src/main/java/com/sequenceiq/it/spark/docker/model/InspectContainerResponse.kt
package com.sequenceiq.it.spark.docker.model

import com.google.gson.annotations.SerializedName

class InspectContainerResponse {

    @SerializedName("Id")
    private var id: String? = null
    @SerializedName("State")
    private var state: ContainerState? = null

    constructor() {
    }

    constructor(id: String) {
        this.id = id
        state = InspectContainerResponse.ContainerState()
        state!!.isRunning = true
    }

    fun setId(id: String) {
        this.id = id
    }

    fun setState(state: ContainerState) {
        this.state = state
    }

    class ContainerState {

        @SerializedName("Running")
        var isRunning: Boolean = false
    }

}
